test/test_utils.h

```diff
@@ -379,47 +317,48 @@ namespace test_utils {
 		// Here and below: Using these functions will cause false-positive CGF diagnostic errors, b/c we are not capturing any accessors.
 		// TODO: For many test cases using these functions it may actually be preferable to circumvent the whole handler mechanism entirely.
 		detail::cgf_diagnostics::teardown();
-		return tm.submit_command_group([&, gs = global_size, go = global_offset](handler& cgh) {
+		auto cg = detail::invoke_command_group_function([&, gs = global_size, go = global_offset](handler& cgh) {
 			cgf(cgh);
 			cgh.parallel_for<KernelName>(gs, go, [](id<KernelDims>) {});
 		});
+		return tm.generate_command_group_task(std::move(cg));
 		detail::cgf_diagnostics::make_available();
 	}
 
 	template <typename KernelName = detail::unnamed_kernel, typename CGF, int KernelDims = 2>
 	detail::task_id add_nd_range_compute_task(detail::task_manager& tm, CGF cgf, celerity::nd_range<KernelDims> execution_range = {{1, 1}, {1, 1}}) {
 		// (See above).
 		detail::cgf_diagnostics::teardown();
-		return tm.submit_command_group([&, er = execution_range](handler& cgh) {
+		auto cg = detail::invoke_command_group_function([&, er = execution_range](handler& cgh) {
 			cgf(cgh);
 			cgh.parallel_for<KernelName>(er, [](nd_item<KernelDims>) {});
 		});
+		return tm.generate_command_group_task(std::move(cg));
 		detail::cgf_diagnostics::make_available();
```

Comment:
Fixing an unreachable code warning now, how did that ever work?!

Reply:
Oops! Well this line not being executed just means that the CGF diagnostics are disabled for the remainder of the test. So it just wouldn't have caught any accessor-misue later on. But I doubt that there's any test that uses both `add_compute_task` et al., as well as a real command group where the diagnostics would matter. In fact, I just removed the call to `teardown` and it looks like we have exactly 1 test that uses `add_compute_task` with a real buffer, producing that warning (all others use mock buffers).

Reply:
Ironicallly, the test in question `get_access can be called on const buffer` does not call `get_access` on a const buffer. I'm refactoring it and splitting it into a RT test and a RT deprecation test to give it some meaning.

Reply:
I've pushed another commit that also moves non-RT sibling tests out to `task_graph_tests`. We should get rid of `add_*_task`, but this will have to be alongside a TDAG generator / TDAG test context refactoring.